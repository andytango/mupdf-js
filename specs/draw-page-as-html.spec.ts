import { forExamplePages, loadExampleFile } from "./helpers/example-file";

describe("drawPageAsHTML", () => {
  forExamplePages("should return a html string", async (page) => {
    const { mupdf, doc } = await loadExampleFile();
    expect(mupdf.drawPageAsHTML(doc, page)).toMatchSnapshot();
  });
});
