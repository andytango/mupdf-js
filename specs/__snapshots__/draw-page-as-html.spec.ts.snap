// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drawPageAsHTML Page [1 / 4] - should return a html string 1`] = `
"<div id="page1" style="width:612.0pt;height:792.0pt">
<p style="top:100.6pt;left:100.0pt;line-height:25.0pt"><b><span style="font-family:Palatino,serif;font-size:25.0pt">Some text with an embedded font!</span></b></p>
<p style="top:127.9pt;left:100.0pt;line-height:18.0pt"><b><span style="font-family:Palatino,serif;font-size:18.0pt">PNG and JPEG images:</span></b></p>
</div>
"
`;

exports[`drawPageAsHTML Page [2 / 4] - should return a html string 1`] = `
"<div id="page2" style="width:612.0pt;height:792.0pt">
<p style="top:100.6pt;left:100.0pt;line-height:25.0pt"><b><span style="font-family:Palatino,serif;font-size:25.0pt">Here is some vector graphics...</span></b></p>
<p style="top:300.6pt;left:100.0pt;line-height:25.0pt"><b><span style="font-family:Palatino,serif;font-size:25.0pt">And here is some wrapped text...</span></b></p>
<p style="top:341.5pt;left:130.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam in</span></p>
<p style="top:356.5pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">suscipit purus. Vestibulum ante ipsum primis in faucibus orci luctus et</span></p>
<p style="top:371.5pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">ultrices posuere cubilia Curae; Vivamus nec hendrerit felis. Morbi</span></p>
<p style="top:386.5pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">aliquam facilisis risus eu lacinia. Sed eu leo in turpis fringilla hendrerit.</span></p>
<p style="top:401.6pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">Ut nec accumsan nisl. Suspendisse rhoncus nisl posuere tortor</span></p>
<p style="top:416.6pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">tempus et dapibus elit porta. Cras leo neque, elementum a rhoncus ut,</span></p>
<p style="top:431.6pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">vestibulum non nibh. Phasellus pretium justo turpis. Etiam vulputate,</span></p>
<p style="top:446.7pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">odio vitae tincidunt ultricies, eros odio dapibus nisi, ut tincidunt lacus</span></p>
<p style="top:461.7pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">arcu eu elit. Aenean velit erat, vehicula eget lacinia ut, dignissim non</span></p>
<p style="top:476.7pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">tellus. Aliquam nec lacus mi, sed vestibulum nunc. Suspendisse</span></p>
<p style="top:491.7pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">potenti. Curabitur vitae sem turpis. Vestibulum sed neque eget dolor</span></p>
<p style="top:506.8pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">dapibus porttitor at sit amet sem. Fusce a turpis lorem. Vestibulum ante</span></p>
<p style="top:521.8pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae;</span></p>
<p style="top:541.8pt;left:130.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">Mauris at ante tellus. Vestibulum a metus lectus. Praesent tempor</span></p>
<p style="top:556.9pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">purus a lacus blandit eget gravida ante hendrerit. Cras et eros metus.</span></p>
<p style="top:571.9pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">Sed commodo malesuada eros, vitae interdum augue semper quis.</span></p>
<p style="top:586.9pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">Fusce id magna nunc. Curabitur sollicitudin placerat semper. Cras et</span></p>
<p style="top:601.9pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">mi neque, a dignissim risus. Nulla venenatis porta lacus, vel rhoncus</span></p>
<p style="top:617.0pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">lectus tempor vitae. Duis sagittis venenatis rutrum. Curabitur tempor</span></p>
<p style="top:632.0pt;left:100.0pt;line-height:13.0pt"><span style="font-family:Arial,sans-serif;font-size:13.0pt">massa tortor.</span></p>
</div>
"
`;

exports[`drawPageAsHTML Page [3 / 4] - should return a html string 1`] = `
"<div id="page3" style="width:612.0pt;height:792.0pt">
<p style="top:100.6pt;left:100.0pt;line-height:25.0pt"><b><span style="font-family:Palatino,serif;font-size:25.0pt">Rendering some SVG paths...</span></b></p>
</div>
"
`;

exports[`drawPageAsHTML Page [4 / 4] - should return a html string 1`] = `
"<div id="page4" style="width:612.0pt;height:792.0pt">
<p style="top:100.6pt;left:100.0pt;line-height:25.0pt"><b><span style="font-family:Palatino,serif;font-size:25.0pt;color:#0000ff">Here is a link!</span></b></p>
<p style="top:152.9pt;left:126.7pt;line-height:16.0pt"><span style="font-family:Chalkboard,serif;font-size:16.0pt">One</span></p>
<p style="top:173.3pt;left:126.7pt;line-height:16.0pt"><span style="font-family:Chalkboard,serif;font-size:16.0pt">Two</span></p>
<p style="top:193.7pt;left:126.7pt;line-height:16.0pt"><span style="font-family:Chalkboard,serif;font-size:16.0pt">Three</span></p>
</div>
"
`;
